packages/components/app/styles/components/app-footer.scss

```
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// app-footer
+//
+
+$app-footer-gap: 24px;
+$app-footer-icon-text-gap: 6px;
+
+.hds-app-footer {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $app-footer-gap;
+  justify-content: flex-end;
+  padding: 24px;
+  color: var(--app-footer-foreground-color);
+  border-top: 1px solid var(--app-footer-border-top-color);
+}
+
+// SUB-ELEMENTS
+
+// hide list if no children
+.hds-app-footer__list:not(:has(li)) { display: none; }
+
+.hds-app-footer__list,
+.hds-app-footer__legal-links {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $app-footer-gap;
+  align-items: center;
+  justify-content: flex-end;
+  width: max-content;
+  max-width: 100%;
```

In a flex context, when content blows up the reason is not the max-width but the lack of an intrinsic initial width; usually the way to fix it is to set `min-width` to `0` (see links in comments I shared the other day)

```suggestion
  min-width: 0;
```